app/views/components/gds/PaginationUtil.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.components.gds
+
+import java.lang.Math.abs
+
+import views.components.gds.PaginationUtil.PaginationItem._
+
+object PaginationUtil {
+
+  def paginationElements(currentPageIndex: Int, pagesTotal: Int, neighbourPagesAmount: Int = 1): List[PaginationItem] = {
+
+    def isCurrentPage(pageNumber: Int): Boolean = pageNumber == currentPageIndex
+
+    if (currentPageIndex < 1)
+      throw new IllegalArgumentException("parameter currentPageIndex must be bigger than 0")
+    if (pagesTotal < 1)
+      throw new IllegalArgumentException("parameter pagesTotal must be bigger than 0")
+    if (currentPageIndex > pagesTotal)
+      throw new IllegalArgumentException("parameter currentPageIndex cannot be bigger than pagesTotal")
+    if (neighbourPagesAmount < 0)
+      throw new IllegalArgumentException("parameter neighbourPagesAmount cannot be negative")
+
+    val neighbourPagesNumbers = ((currentPageIndex - neighbourPagesAmount) to (currentPageIndex + neighbourPagesAmount))
+      .filter(pageNumber => pageNumber > 0 && pageNumber <= pagesTotal)
+
+    val leadingElements: List[PaginationItem] = abs(neighbourPagesNumbers.head - 1) match {
```

Comment:
Is it possible that head will be empty?

What do you think about `.headOption.getOrElse()`?

Reply:
I really doubt it since you have 4 parameters checks at the beginning of the method, so this list will be at least one element long.
But let me investigate how to make more error safe.